File: 001_Pilhas/Ex03.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace _001_Pilhas
{
    public partial class Ex03 : Form
    {
        public Ex03()
        {
            InitializeComponent();
        }
        // COPIAR A PARTIR DAQUI

        const int MAX = 50;
        struct tp_pilha
        {
            public int topo;
            public int[] vetor;
        }
        tp_pilha pilha;
        void Inicializa(ref tp_pilha p)
        {
            p.topo = -1;
            p.vetor = new int[MAX];
        }
        bool EstaVazia(tp_pilha p)
        {
            return (p.topo == -1);
        }
        bool EstaCheia(tp_pilha p)
        {  //verificar se esta cheia
            return (p.topo == MAX - 1);
        }
        // Insere um elemento
        void Empilha(ref tp_pilha p, int valor)
        {
            p.vetor[++p.topo] = valor;
        }
        // Remove um elemento
        int Desempilha(ref tp_pilha p)
        {
            return (p.vetor[p.topo--]);
        }

        private void BProcessa_Click(object sender, EventArgs e)
        {
            int NumInt, i = 1;
            string NumResultado = "";
            string Oper = TBOperations.Text;
            NumInt = Convert.ToInt32(TBNElements.Text);

            Inicializa(ref pilha);

            foreach (char l in Oper)
            {
                if (l == 'E')
                {
                    Empilha(ref pilha, i);
                    i++;
                 }
                else if (l == 'D')
                    NumResultado += Convert.ToString(Desempilha(ref pilha));
            }
            TBNumResultado.Text = NumResultado;
        }

    }
}

File: python/ex03.py
class Pilha(object):

    def __init__(self, topo=-1, max=50):
        self.topo = topo
        self.vetor = []
        self.max = max
        
    def EstaVazia(self):
        return (self.topo == -1)
    
    def EstaCheia(self):
        return (self.topo == max - 1)

    # Insere um elemento
    def Empilha(self, valor):
        self.vetor.append(valor)
        self.topo += 1

    # Remove um elemento
    def Desempilha(self):
        v = self.vetor[self.topo]
        self.vetor.pop()
        self.topo -= 1
        return v


p = Pilha()
NumInt = int(input("Número: "))
Oper = input("OPERAÇÕES: ")

i = 1
NumResultado = ""
for l in Oper:
    if (l == 'E'):
        p.Empilha(i)
        i += 1
    elif (l == 'D'):
        NumResultado += str(p.Desempilha())
print(NumResultado)